ConsoleGraphics/ConsoleProgressBar.cs

```diff
@@ -69,18 +93,41 @@ public void Draw(uint currentUnitOfWork)
                 throw new ArgumentOutOfRangeException("currentUnitOfWork", "currentUnitOfWork must be less than TotalUnitsOfWork");
             }
 
+            if (ShowETA && !StartTime.HasValue)
+            {
+                throw new Exception("Need to call StartWork if want to calculate the ETA");
+            }
```

Comment:
Consider capturing the start time on the first call to Draw. This would let you remove the StartWork method.

Reply:
That's an option but this lead of missing the information of the duration of the very first unit of work....  

Ideally you should create the progress bar, declare to START the work...  doing actually the first unit of work..  and call the Draw method...   

The way i've implemented consider even the very first unit of work timespan...